src/app/services/wallet.service.ts

```diff
@@ -121,73 +164,106 @@ export class WalletService {
         },
-        error => {
-          console.log("Error", error);
+        (error) => {
+          alert(`syncWalletList error ${error}`);
         }
       );
   }
 
-  checkWalletBalance(walletId: string, passphrase: string): Observable<string> {
+  checkWalletBalance(
+    walletId: string,
+    passphrase: string,
+    enckey: string
+  ): Observable<string> {
     return this.http.post<string>(this.coreUrl, {
       jsonrpc: "2.0",
       id: "jsonrpc",
       method: "wallet_balance",
       params: [
         {
           name: walletId,
-          passphrase: _.isNil(passphrase) ? "" : passphrase
-        }
-      ]
+          passphrase: _.isNil(passphrase) ? "" : passphrase,
+          enckey: _.isNil(passphrase) ? "" : enckey,
+        },
+      ],
     });
   }
-  checkWalletAddress(walletId: string, passphrase: string): Observable<string> {
+  checkWalletAddress(
+    walletId: string,
+    passphrase: string,
+    enckey: string
+  ): Observable<string> {
     return this.http.post<string>(this.coreUrl, {
       jsonrpc: "2.0",
       id: "jsonrpc",
       method: "wallet_listTransferAddresses",
       params: [
         {
           name: walletId,
-          passphrase: _.isNil(passphrase) ? "" : passphrase
-        }
-      ]
+          passphrase: _.isNil(passphrase) ? "" : passphrase,
+          enckey: _.isNil(passphrase) ? "" : enckey,
+        },
+      ],
     });
   }
 
-  checkWalletViewKey(walletId: string, passphrase: string): Observable<string> {
+  checkWalletEncKey(walletId: string, passphrase: string): Observable<string> {
+    return this.http.post<string>(this.coreUrl, {
+      jsonrpc: "2.0",
+      id: "jsonrpc",
+      method: "wallet_getEncKey",
+      params: [
+        {
+          name: walletId,
+          passphrase: _.isNil(passphrase) ? "" : passphrase,
+        },
+      ],
+    });
+  }
+
+  checkWalletViewKey(
+    walletId: string,
+    passphrase: string,
+    enckey: string
+  ): Observable<string> {
     return this.http.post<string>(this.coreUrl, {
       jsonrpc: "2.0",
       id: "jsonrpc",
       method: "wallet_getViewKey",
       params: [
         {
           name: walletId,
-          passphrase: _.isNil(passphrase) ? "" : passphrase
-        }
-      ]
+          passphrase: _.isNil(passphrase) ? "" : passphrase,
+          enckey: _.isNil(passphrase) ? "" : enckey,
+        },
+      ],
     });
   }
 
-  checkWalletTxnHistory(walletId: string, passphrase: string) {
+  checkWalletTxnHistory(walletId: string, passphrase: string, enckey: string) {
     return this.http.post<string>(this.coreUrl, {
       jsonrpc: "2.0",
       id: "jsonrpc",
       method: "wallet_transactions",
       params: [
         {
           name: walletId,
-          passphrase: _.isNil(passphrase) ? "" : passphrase
-        }
-      ]
+          passphrase: _.isNil(passphrase) ? "" : passphrase,
+          enckey: _.isNil(passphrase) ? "" : enckey,
+        },
+        0,
+        1000,
```

Comment:
Can put a TODO here later we can add pagination and records per page options

Reply:
ok